### PR TITLE
Using run-p command to execute all tasks of build-dist in parallel

### PR DESCRIPTION
**Pull request by zhangyiatmicrosoft**
## Launch Checklist

- Before it took ~150sec and after: ~40s. In CI machines, packaging now can finish <5m, whereas before > 10m.
- Added timer-start and timer-stop which compute elapsed time of any npm task so we can use it in the future.

 - [x] Confirm **your changes do not include backports from Mapbox projects** (unless with compliant license) - if you are not sure about this, please ask!
 - [x] Briefly describe the changes in this PR.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -18,6 +18,7 @@
 - Adding a `warnonce` when terrain and hillshade source are the same ([#2298](https://github.com/maplibre/maplibre-gl-js/pull/2298))
 - Remove a deprecation warning by removing an empty texture that is no longer being used in the codebase ([#2299](https://github.com/maplibre/maplibre-gl-js/pull/2299))
 - Improve initial loading performance by lazy serializing layers only when needed. ([#2306](https://github.com/maplibre/maplibre-gl-js/pull/2306))
+- Improve "build-dist" by running its taks in parallel. ([#2349](https://github.com/maplibre/maplibre-gl-js/pull/2349))
```
Comment (HarelM) [inline CHANGELOG.md:21]: Typo... Also not sure this is interesting enough to be in the changelog.

File: package.json [hunk 2]
```diff
@@ -172,7 +172,9 @@
     "benchmark": "ts-node test/bench/run-benchmarks.ts",
     "gl-stats": "ts-node test/bench/gl-stats.ts",
     "prepare": "npm run generate-dist-package && npm run codegen",
-    "typecheck": "tsc --noEmit"
+    "typecheck": "tsc --noEmit",
+    "timer-start": "ts-node ./timer-start.ts",
```
Comment (HarelM) [inline package.json:176]: Feels like an overkill to add this...
CI have this out of the box I believe.
I would only keep the run-p code as part of this PR.

Comment (zhangyiatmicrosoft) [reply package.json]: Not on local machine though. Planning to do more optimizations of build tasks. A timer is very handy.

**Comment by HarelM** [comment]
I don't think this extra complexity is needed for everyone. Let's keep the optimization results without the extra files and steps.
You can have it in a branch for your personal use.
We did a lot of work to simplify the build process and reduced complexity, I don't want to introduce steps that are not mandatory.

**Comment by zhangyiatmicrosoft** [comment]
> I don't think this extra complexity is needed for everyone. Let's keep the optimization results without the extra files and steps. You can have it in a branch for your personal use. We did a lot of work to simplify the build process and reduced complexity, I don't want to introduce steps that are not mandatory.

ok

**Review by HarelM — APPROVED** [review]

**Pull request merged**